internal/source/git.go

```diff
@@ -231,10 +237,37 @@ func (f *billyFS) ReadDir(name string) ([]fs.DirEntry, error) {
 
 type billyFile struct {
 	billy.File
-	fi    os.FileInfo
-	fiErr error
+	fi os.FileInfo
 }
 
 func (b billyFile) Stat() (fs.FileInfo, error) {
-	return b.fi, b.fiErr
+	return b.fi, nil
+}
+
+func (b billyFile) Close() error {
+	if b.File == nil {
+		return nil
+	}
+	return b.File.Close()
+}
+
+func (b billyFile) Read(data []byte) (int, error) {
+	if b.File != nil {
+		return b.File.Read(data)
+	}
+	return 0, fs.ErrNotExist
```

Comment:
Sorry for all the back and forth on this. I'm learning this at the same time you are...

`fs.ErrNotExist` would be returned by `Stat()`. So by the time we have a `billyFile` we know _something_ exists at the opened path.

[Seems like](https://go.dev/play/p/ycgOLd_n7dQ) reading an OS directory results in:
```go
return 0, &PathError{Op:"read", Path: path, Err: syscall.EISDIR}
```

And this makes me think that we need to check `billyFile.fi.IsDir()` during `Read`, and if we're doing _that_, then maybe what we want is just a custom `Read()` on `billyDirFile`:

```go
func (b billyDirFile) Read(data []byte) (int, error) {
	return 0, &PathError{Op:"read", Path: path, Err: syscall.EISDIR}
}
```

and then get rid of the `billyFile` `Read()` altogether (so that we just use the underlying `billy.File.Read()` as we were before)

Reply:
OK.  Updated.